wgengine/magicsock/derp.go

```diff
@@ -783,6 +783,9 @@ func (c *Conn) SetDERPMap(dm *tailcfg.DERPMap) {
 		return
 	}
 
+	// Provide the DERP map to the health package, which can use it when generating bad health notifications.
+	c.health.SetDERPMap(dm)
```

Comment:
we don't necessarily need or want to do this while also holding c.mu.

move this up earlier?

but then the question is why magicsock has this job of telling health? maybe the caller that got the new derpmap originally and told everybody else should be telling health.

Reply:
moved to `SetControlClientStatus` in `local.go` which appears to be the caller of this. Not sure if there is a better place...